universe/ValueRefs.cpp

```diff
@@ -285,20 +286,18 @@ std::string ValueRefBase::InvariancePattern() const {
         .append(ConstantExpr()                  ? "C" : "c");
 }
 
-MeterType NameToMeter(const std::string& name) {
-    MeterType retval = MeterType::INVALID_METER_TYPE;
-    auto it = GetMeterNameMap().find(name);
-    if (it != GetMeterNameMap().end())
-        retval = it->second;
-    return retval;
+MeterType NameToMeter(const std::string_view name) {
+    for (int i = 0; i < static_cast<int>(NAME_BY_METER.size()); i++)
+    {
+        if (NAME_BY_METER[i] == name)
+            return static_cast<MeterType>(i - 1);
+    }
+
+    return MeterType::INVALID_METER_TYPE;
 }
 
-const std::string& MeterToName(MeterType meter) {
-    for (auto& [name, type] : GetMeterNameMap()) {
-        if (type == meter)
-            return name; // CodeQL reports "Returning stack-allocated memory", but the name structured binding should be a reference as far as I can tell
-    }
-    return EMPTY_STRING;
+const std::string_view MeterToName(const MeterType meter) {
+    return NAME_BY_METER[static_cast<std::underlying_type_t<MeterType>>(meter) + 1];
```

Comment:
I'd add a comment here noting that the lowest value for `meter` should be `INVALID_METER_TYPE = -1` and the highest less than `NUM_METER_TYPES`

Reply:
Done 👍🏼